pkg/ccl/importccl/read_import_pgdump.go

```diff
@@ -299,83 +299,161 @@ func readPostgresCreateTable(
+	stmt interface{},
+) error {
+	switch stmt := stmt.(type) {
+	case *tree.CreateTable:
+		name, err := getTableName(&stmt.Table)
+		if err != nil {
+			return err
+		}
+		if match != "" && match != name {
+			createTbl[name] = nil
+		} else {
+			createTbl[name] = stmt
+		}
+	case *tree.CreateIndex:
+		name, err := getTableName(&stmt.Table)
+		if err != nil {
+			return err
+		}
+		create := createTbl[name]
+		if create == nil {
+			break
+		}
+		var idx tree.TableDef = &tree.IndexTableDef{
+			Name:        stmt.Name,
+			Columns:     stmt.Columns,
+			Storing:     stmt.Storing,
+			Inverted:    stmt.Inverted,
+			Interleave:  stmt.Interleave,
+			PartitionBy: stmt.PartitionBy,
+		}
+		if stmt.Unique {
+			idx = &tree.UniqueConstraintTableDef{IndexTableDef: *idx.(*tree.IndexTableDef)}
+		}
+		create.Defs = append(create.Defs, idx)
+	case *tree.AlterTable:
+		name, err := getTableName2(stmt.Table)
+		if err != nil {
+			return err
+		}
+		create := createTbl[name]
+		if create == nil {
+			break
+		}
+		for _, cmd := range stmt.Cmds {
+			switch cmd := cmd.(type) {
+			case *tree.AlterTableAddConstraint:
+				switch con := cmd.ConstraintDef.(type) {
+				case *tree.ForeignKeyConstraintTableDef:
+					if !fks.skip {
+						tableFKs[name] = append(tableFKs[name], con)
+					}
+				default:
+					create.Defs = append(create.Defs, cmd.ConstraintDef)
+				}
+			case *tree.AlterTableSetDefault:
+				for i, def := range create.Defs {
+					def, ok := def.(*tree.ColumnTableDef)
+					if !ok || def.Name != cmd.Column {
+						continue
+					}
+					def.DefaultExpr.Expr = cmd.Default
+					create.Defs[i] = def
+				}
+			case *tree.AlterTableAddColumn:
+				if cmd.IfNotExists {
+					return errors.Errorf("unsupported statement: %s", stmt)
+				}
+				create.Defs = append(create.Defs, cmd.ColumnDef)
+			case *tree.AlterTableValidateConstraint:
+				// ignore
+			default:
+				return errors.Errorf("unsupported statement: %s", stmt)
 			}
-			for _, cmd := range stmt.Cmds {
-				switch cmd := cmd.(type) {
-				case *tree.AlterTableAddConstraint:
-					switch con := cmd.ConstraintDef.(type) {
-					case *tree.ForeignKeyConstraintTableDef:
-						if !fks.skip {
-							tableFKs[name] = append(tableFKs[name], con)
+		}
+	case *tree.CreateSequence:
+		name, err := getTableName(&stmt.Name)
+		if err != nil {
+			return err
+		}
+		if match == "" || match == name {
+			createSeq[name] = stmt
+		}
+	// Some SELECT statements mutate schema. Search for those here. If it is not exactly a SELECT that mutates
+	// schema, ignore it.
+	case *tree.Select:
+		switch sel := stmt.Select.(type) {
+		case *tree.SelectClause:
+			for _, selExpr := range sel.Exprs {
+				switch expr := selExpr.Expr.(type) {
+				case *tree.FuncExpr:
+					// Look for function calls that mutate schema (this is actually a thing).
+					semaCtx := tree.MakeSemaContext()
+					_, err := expr.TypeCheck(ctx, &semaCtx, nil /* desired */)
+					if err != nil {
+						return err
```

Comment:
Personal preference (take with a grain of salt :) ), I would condense this to 
`
if _, err := expr.TypeCheck(ctx, &semaCtx, nil); err != nil {
  return err
}
`

Reply:
done